### PR TITLE
Feature/lau 669 configure functional test

### PR DESCRIPTION
**Pull request by mhisltd**
**Before creating a pull request make sure that:**

- [X] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)
- [ ] the `keep-helm` label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/resources/application-functional.yaml [hunk 1]
```diff
@@ -28,7 +28,41 @@ api-version-number: 1
...
+  api:
+    url: ${IDAM_API_URL:http://localhost:5000}
+  client:
+    id: ${IDAM_CLIENT_NAME:lau}
+    secret: ${IDAM_CLIENT_SECRET:12345678}
+    redirect_uri: ${IDAM_CLIENT_REDIRECT_URI:http://localhost:3451/oauth2redirect}
+
+
+  grant:
+    type:authorised.services: idam_api,lau_frontend,idam-user-disposer
```
Comment (MartinYSpasov) [inline src/functionalTest/resources/application-functional.yaml:43]: Do we need type?

File: src/functionalTest/resources/application-functional.yaml [hunk 2]
```diff
@@ -28,7 +28,41 @@ api-version-number: 1
...
+    type:api.url: http://localhost:4550
+client.secret: ${IDAM_CLIENT_SECRET}
+
+spring:
+  datasource:
+    url: jdbc:postgresql://${LAU_IDAM_DB_HOST:localhost}:${LAU_IDAM_DB_PORT:5054}/${LAU_IDAM_DB_NAME:lau_idam}${LAU_IDAM_DB_CONN_OPTIONS:}
+    password: ${LAU_IDAM_DB_PASSWORD:laupass}
+
+s2s.url: http://rpe-service-auth-provider-aat.service.core-compute-aat.internal/testing-support
+logon.endpoint: /audit/logon
```
Comment (MartinYSpasov) [inline src/functionalTest/resources/application-functional.yaml:59]: The endpoint can be constant.

File: src/functionalTest/resources/application-functional.yaml [hunk 3]
```diff
@@ -28,7 +28,41 @@ api-version-number: 1
...
+idam.grant.type: password
+idam.username: lautest@test.com
+idam.password: Password12
+idam.redirect.url: https://lau-case.aat.platform.hmcts.net/oauth2/callback
+idam.token.url: https://idam-api.aat.platform.hmcts.net/o/token
+api.url: TEST_URL
```
Comment (MartinYSpasov) [inline src/functionalTest/resources/application-functional.yaml:65]: What is this?

File: src/functionalTest/resources/application-functional.yaml [hunk 4]
```diff
@@ -28,7 +28,41 @@ api-version-number: 1
...
+
+    type:authorised.roles: cft-audit-investigator
+
+
+    type:db.allow.delete.record: ${DB_ALLOW_DELETE_RECORD:true}
+
+    type:api.url: http://localhost:4550
+client.secret: ${IDAM_CLIENT_SECRET}
+
+spring:
```
Comment (MartinYSpasov) [inline src/functionalTest/resources/application-functional.yaml:53]: Do you need datasource?

**Review by MartinYSpasov — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/laubackend/idam/serenityfunctionaltests/helper/PropertyReader.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            properties = new Properties();
+            properties.load(PropertyReader.class.getResourceAsStream("/application-functional.yaml"));
+        } catch (final IOException ioException) {
+            log.error(ioException.getMessage(), ioException);
+        }
+    }
+
+    public static PropertyReader getInstance() {
+
+        synchronized (PropertyReader.class) {
```
[the remark this answers is not in the document]
Comment (mhisltd) [reply src/functionalTest/java/uk/gov/hmcts/reform/laubackend/idam/serenityfunctionaltests/helper/PropertyReader.java:26]: There was an issue highlighted by build for non safe threading thats the reason to use it.

File: src/functionalTest/java/uk/gov/hmcts/reform/laubackend/idam/serenityfunctionaltests/config/EnvConfig.java [hunk 6]
```diff
@@ -2,10 +2,8 @@
 
 public final class EnvConfig {
 
-    public static final String API_URL = System.getenv("TEST_URL");
     public static final String IDAM_CLIENT_SECRET = System.getenv("IDAM_CLIENT_SECRET");
```
[the remark this answers is not in the document]
Comment (mhisltd) [reply src/functionalTest/java/uk/gov/hmcts/reform/laubackend/idam/serenityfunctionaltests/config/EnvConfig.java:5]: Done

File: src/functionalTest/java/uk/gov/hmcts/reform/laubackend/idam/serenityfunctionaltests/helper/AuthorizationHeaderHelper.java [hunk 7]
```diff
@@ -14,32 +14,27 @@
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 import static uk.gov.hmcts.reform.laubackend.idam.serenityfunctionaltests.config.EnvConfig.IDAM_CLIENT_SECRET;
 import static uk.gov.hmcts.reform.laubackend.idam.serenityfunctionaltests.utils.TestConstants.CLIENT_ID;
```
[the remark this answers is not in the document]
Comment (mhisltd) [reply src/functionalTest/java/uk/gov/hmcts/reform/laubackend/idam/serenityfunctionaltests/helper/AuthorizationHeaderHelper.java:16]: Done

File: src/functionalTest/resources/application-functional.yaml [hunk 8]
```diff
@@ -1,4 +1,4 @@
-targetInstance: ${TEST_URL:http://localhost:4550}
+targetInstance: ${TEST_URL:http//localhost:4550}
```
[the remark this answers is not in the document]
Comment (mhisltd) [reply src/functionalTest/resources/application-functional.yaml:1]: Typo, restored.

File: src/functionalTest/resources/application-functional.yaml (hunk 2 above)
Comment (mhisltd) [reply src/functionalTest/resources/application-functional.yaml:59]: Removed.

File: src/functionalTest/resources/application-functional.yaml (hunk 3 above)
Comment (mhisltd) [reply src/functionalTest/resources/application-functional.yaml:65]: Application under test.

File: src/functionalTest/resources/application-functional.yaml [hunk 9]
```diff
@@ -28,7 +28,41 @@ api-version-number: 1
...
+  grant:
+    type:authorised.services: idam_api,lau_frontend,idam-user-disposer
+
+    type:authorised.roles: cft-audit-investigator
+
+
+    type:db.allow.delete.record: ${DB_ALLOW_DELETE_RECORD:true}
+
+    type:api.url: http://localhost:4550
+client.secret: ${IDAM_CLIENT_SECRET}
```
[the remark this answers is not in the document]
Comment (mhisltd) [reply src/functionalTest/resources/application-functional.yaml:51]: I've now replaced it with the actual value, we cannot use it in the same fashion as before as the property reader reads things differently.

File: src/functionalTest/resources/application-functional.yaml (hunk 1 above)
Comment (mhisltd) [reply src/functionalTest/resources/application-functional.yaml:43]: Removed

File: src/functionalTest/resources/application-functional.yaml (hunk 4 above)
Comment (mhisltd) [reply src/functionalTest/resources/application-functional.yaml:53]: Removed.

File: src/functionalTest/resources/application-functional.yaml [hunk 10]
```diff
@@ -28,7 +28,25 @@ api-version-number: 1
...
+  client:
+    id: ${IDAM_CLIENT_NAME:lau}
+    secret: ${IDAM_CLIENT_SECRET:12345678}
+    redirect_uri: ${IDAM_CLIENT_REDIRECT_URI:http://localhost:3451/oauth2redirect}
+
+s2s.url: http://rpe-service-auth-provider-aat.service.core-compute-aat.internal/testing-support
+idam.grant.type: password
+idam.username: lautest@test.com
+idam.password: Password12
+idam.client.secret: gQ4pHYQ8TTYdQp8d
```
Comment (MartinYSpasov) [inline src/functionalTest/resources/application-functional.yaml:45]: This value should not be public!

File: src/functionalTest/java/uk/gov/hmcts/reform/laubackend/idam/serenityfunctionaltests/helper/PropertyReader.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.laubackend.idam.serenityfunctionaltests.helper;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.Properties;
+
+@SuppressWarnings({"PMD"})
+@Slf4j
```
[the remark this answers is not in the document]
Comment (mhisltd) [reply src/functionalTest/java/uk/gov/hmcts/reform/laubackend/idam/serenityfunctionaltests/helper/PropertyReader.java:9]: There is logging on line 21.

File: src/functionalTest/resources/application-functional.yaml (hunk 10 above)
Comment (mhisltd) [reply src/functionalTest/resources/application-functional.yaml]: Removed

File: src/functionalTest/java/uk/gov/hmcts/reform/laubackend/idam/serenityfunctionaltests/helper/PropertyReader.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.laubackend.idam.serenityfunctionaltests.helper;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.Properties;
+
+@SuppressWarnings({"PMD"})
```
[the remark this answers is not in the document]
Comment (mhisltd) [reply src/functionalTest/java/uk/gov/hmcts/reform/laubackend/idam/serenityfunctionaltests/helper/PropertyReader.java]: Dpme

**Pull request merged**